third_party/CMakeLists.txt

```diff
@@ -63,11 +63,9 @@ add_subdirectory(glfw)
 add_subdirectory(glm)
 
 #### VulkanMemoryAllocator
-include_directories(VulkanMemoryAllocator/src)
-include_directories(${Vulkan_INCLUDE_DIRS})
 add_library(VulkanMemoryAllocator STATIC "VulkanMemoryAllocatorBuild.cpp")
-target_include_directories(VulkanMemoryAllocator INTERFACE VulkanMemoryAllocator/src)
-target_include_directories(VulkanMemoryAllocator PRIVATE "${SCIN_EXT_INSTALL_DIR}/include")
+target_include_directories(VulkanMemoryAllocator PUBLIC VulkanMemoryAllocator/src)
```

Comment:
`PUBLIC` = `INTERFACE` && `PRIVATE`

Reply:
So the memory allocator needs the vulkan SDK to compile itself. I thought PRIVATE was telling cmake to provide the path the vulkan SDK header to that library but not transitively provide that path to dependent libraries.

Reply:
you had it set to INTERFACE before, which tells cmake to _only_ transitively provide this include directory, but not use it when compiling VulkanMemoryAllocator itself. it looks like scinsynth depends on these headers too, though -- if I set this to PRIVATE, then i see this:

```
In file included from ../src/scinsynth.cpp:13:
../src/vulkan/Buffer.hpp:6:10: fatal error: vk_mem_alloc.h: No such file or directory
    6 | #include "vk_mem_alloc.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
That's correct. The Vulkan 1.0 standard, which we are limited to for the moment because the MacOS Vulkan emulation only supports 1.0, allows for a very small number of allocated objects. Best practice is to do a single memory allocation and BYO memory allocator, and we're using this library for exactly that.

Reply:
makes sense! i'm just pointing out that the memory allocator lib isn't the only part of this project that requires these headers.